tools/nnyeah/tests/utils/Compiler.cs

```diff
@@ -75,13 +76,38 @@ static string PlatformLibPath (PlatformName platformName, string libName)
 				_ => throw new NotImplementedException (),
 			};
 
-		static string XamarinLibName (PlatformName platformName) =>
+		public static string XamarinLibName (PlatformName platformName) =>
 			platformName switch {
 				PlatformName.macOS => "Xamarin.Mac",
 				PlatformName.iOS => "Xamarin.iOS",
 				PlatformName.tvOS => "Xamarin.TVOS",
 				PlatformName.watchOS => "Xamarin.WatchOS",
 				_ => throw new NotImplementedException (),
 			};
+
+		public static string XamarinPlatformLibraryPath (PlatformName platformName) =>
+			Path.Combine (XamarinPlatformLibDirectory (platformName), $"{XamarinLibName (platformName)}.dll");
+
+		public static string MicrosoftPlatformLibDirectory (PlatformName platformName) =>
```

Comment:
You should be able to use the reference assemblies:

https://github.com/xamarin/xamarin-macios/blob/92eda7f35399f442f8730aa539571ad8fd44d7da/tests/common/Configuration.cs#L514-L521